app/assets/src/components/ui/controls/dropdowns/BareDropdown.jsx

```diff
@@ -111,6 +113,20 @@ class BareDropdown extends React.Component {
     }
   };
 
+  handleMenuClick = e => {
+    const { closeOnClick, search } = this.props;
+
+    // When search is true, semantic-ui doesn't close the dropdown when the menu is clicked
+    // for some reason. Manually close it in this case.
+    if (search && this.baseDropdownRef.current) {
```

Comment:
Should we also check for `closeOnClick` on this if?
Not sure I fully understand it... but what would happen on a multiple selection with search, for instance? Like the metadata selector? Does a click in the option go through here?